### PR TITLE
add cherryb studio cb1800 keyboard

### PR DESCRIPTION
**Pull request by duongaanh**
Add CherryB Studio CB 1800 Keyboard PCB
## Description

A G80-1800 inspired PCB for CB1800 Zakabato and any suitable 1800 layout keyboards with many configurable layouts

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cherrybstudio/cb1800/rules.mk [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+
+LAYOUTS = all
```
Comment (drashna) [inline keyboards/cherrybstudio/cb1800/rules.mk:24]: `LAYOUT_all` shouldn't be added here, as it's a catch all for boards that don't have a specific community layout, or that have multiple layouts. 

```suggestion
```

File: keyboards/cherrybstudio/cb1800/info.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+    "keyboard_name": "cb1800",
+    "url": "https://discord.gg/qVwv3gcq83",
+    "maintainer": "duongaanh",
+    "width": 19.5,
+    "height": 7.25,
+    "layouts": {
+        "LAYOUT_winkey": {
```
Comment (drashna) [inline keyboards/cherrybstudio/cb1800/info.json:8]: This needs to match layout macros in the h file (eg, `LAYOUT_all`)

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/cherrybstudio/cb1800/info.json (hunk 2 above)
Comment (duongaanh) [reply keyboards/cherrybstudio/cb1800/info.json:8]: oh, thanks. But I have a question that my PCB got so many configurable layout, do I need to create every layout combination for the info.json? it's will make so many layouts.

File: keyboards/cherrybstudio/cb1800/config.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4342 // CB
+#define PRODUCT_ID      0x1818
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    CherryB.Studio
+#define PRODUCT         CherryB.Studio CB1800
+#define DESCRIPTION     CherryB.Studio CB1800
```
Comment (drashna) [inline keyboards/cherrybstudio/cb1800/config.h:24]: Deprecated, should be removed. ```suggestion
```

File: keyboards/cherrybstudio/cb1800/config.h [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ *
+ *                        0   1   2   3   4   5   6   7   8   9
+*/
+#define MATRIX_ROW_PINS { B0, B1, B2, B3, B4, B5, B6, B7, C6, C7 }
+#define MATRIX_COL_PINS { D0, D1, D2, D3, D4, D5, D6, D7, F0, F1, F4 }
+//#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+#define LED_CAPS_LOCK_PIN F5
```
Comment (drashna) [inline keyboards/cherrybstudio/cb1800/config.h:47]: Just a formatting nit pick. 
```suggestion

/* LED Indicators *?
#define LED_CAPS_LOCK_PIN F5
```

File: keyboards/cherrybstudio/cb1800/config.h [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+#define DEBOUNCE 5
+
+#define RGB_DI_PIN E6
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 20 
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#endif
```
Comment (drashna) [inline keyboards/cherrybstudio/cb1800/config.h:62]:
```suggestion
#ifdef RGB_DI_PIN
#    define RGBLIGHT_ANIMATIONS
#    define RGBLED_NUM 20 
#    define RGBLIGHT_HUE_STEP 8
#    define RGBLIGHT_SAT_STEP 8
#    define RGBLIGHT_VAL_STEP 8
#endif
```

File: keyboards/cherrybstudio/cb1800/info.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+        },
```
Comment (drashna) [inline keyboards/cherrybstudio/cb1800/info.json:13]: This should be removed, as well, if there is no matching layout in the h file. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/cherrybstudio/cb1800/readme.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# CherryB Studio CB1800
+
+![cb1800](https://i.imgur.com/QRJ6DFs.jpg)
```
Comment (drashna) [inline keyboards/cherrybstudio/cb1800/readme.md:3]:
```suggestion
![cb1800](https://i.imgur.com/QRJ6DFsl.jpg)
```

**Review by drashna — APPROVED** [review]

File: keyboards/cherrybstudio/cb1800/config.h [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4342 // CB
+#define PRODUCT_ID      0x1818
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    CherryB.Studio
+#define PRODUCT         CherryB.Studio CB1800
```
Comment (fauxpark) [inline keyboards/cherrybstudio/cb1800/config.h:23]:
```suggestion
#define PRODUCT         CB1800
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**